tests/unit/backend/services/webhook.service.spec.ts

```diff
@@ -0,0 +1,151 @@
+  const MOCK_SIGNATURE = 'mock-signature'
+  const MOCK_WEBHOOK_URL = '/webhook-endpoint'
+  const MOCK_NOW = Date.now()
+  const ERROR_MSG = 'test-message'
+  const MOCK_SUCCESS_RESPONSE = {
+    data: {
+      result: 'test-result',
+    },
+    status: 200,
+    statusText: 'success',
+    headers: {},
+    config: {},
+  }
+  const MOCK_STRINGIFIED_PARSED_RESPONSE = {
+    data: '{"result":"test-result"}',
+    status: 200,
+    statusText: 'success',
+    headers: '{}',
+  }
+
+  beforeEach(async () => {
+    const preloaded = await dbHandler.insertFormCollectionReqs({
+      userId: MOCK_ADMIN_OBJ_ID,
+    })
+    testEncryptForm = preloaded.encryptForm
+    testEncryptSubmission = preloaded.encryptSubmission
+    testSubmissionWebhookView = testEncryptSubmission.getWebhookView()
+    testWebhookParam = {
+      webhookUrl: MOCK_WEBHOOK_URL,
+      submissionWebhookView: testSubmissionWebhookView,
+      submissionId: testEncryptSubmission._id,
+      formId: testEncryptForm._id,
+      now: MOCK_NOW,
+      signature: MOCK_SIGNATURE,
+    }
+    testConfig = {
+      headers: {
+        'X-FormSG-Signature': `t=${MOCK_NOW},s=${testEncryptSubmission._id},f=${testEncryptForm._id},v1=${MOCK_SIGNATURE}`,
+      },
+      maxRedirects: 0,
+    }
+  })
+
+  function delay(delayInms) {
+    return new Promise((resolve) => {
+      setTimeout(() => {
+        resolve()
+      }, delayInms)
+    })
+  }
+
+  describe('handleWebhookFailure', () => {
+    it('should update submission document with failed webhook response', async () => {
+      // Act
+      const error = new AxiosError(ERROR_MSG, MOCK_SUCCESS_RESPONSE)
+      handleWebhookFailure(error, testWebhookParam)
+      await delay(3000)
+      // Assert
+      let submission = await EncryptSubmission.findById(
+        testEncryptSubmission._id,
+      )
+      expect(submission.webhookResponses[0]).toEqual(
+        expect.objectContaining({
+          webhookUrl: MOCK_WEBHOOK_URL,
+          signature: MOCK_SIGNATURE,
+          errorMessage: ERROR_MSG,
+          response: expect.objectContaining(MOCK_STRINGIFIED_PARSED_RESPONSE),
+        }),
+      )
+    })
+    it('should update submission document with failed webhook validation', async () => {
+      // Act
+      const error = new WebhookValidationError(ERROR_MSG)
+      handleWebhookFailure(error, testWebhookParam)
+      await delay(3000)
+      // Assert
+      let submission = await EncryptSubmission.findById(
+        testEncryptSubmission._id,
+      )
+      expect(submission.webhookResponses[0]).toEqual(
+        expect.objectContaining({
+          webhookUrl: MOCK_WEBHOOK_URL,
+          signature: MOCK_SIGNATURE,
+          errorMessage: ERROR_MSG,
+        }),
+      )
+    })
+  })
+
+  describe('handleWebhookSuccess', () => {
+    it('should update submission document with successful webhook response', async () => {
+      // Act
+      handleWebhookSuccess(MOCK_SUCCESS_RESPONSE, testWebhookParam)
```

Comment:
Same here. We can always make this a promise, the server does not need to await it. Basically just for testing.

Reply:
will do